internal/controller/stas/scan_job_controller.go

```diff
@@ -53,6 +63,89 @@ func (r *ScanJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
 				ignoreDeletionPredicate(),
 			)).
 		Complete(r.reconcile())
+	if err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("backOffScanJobPod").
+		Watches(
+			&source.Kind{Type: &eventsv1.Event{}},
+			handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+				e := obj.(*eventsv1.Event)
+				return []reconcile.Request{
+					{NamespacedName: types.NamespacedName{
+						Name:      e.Regarding.Name,
+						Namespace: e.Regarding.Namespace,
+					}},
+				}
+			}),
+			builder.WithPredicates(
+				inNamespacePredicate(r.ScanJobNamespace),
+				eventRegardingKind("Pod"),
+				eventReason("BackOff"),
+			),
+		).
+		Complete(r.reconcileBackOffJobPod())
+}
+
+func (r *ScanJobReconciler) reconcileBackOffJobPod() reconcile.Func {
+	return func(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+		logf.FromContext(ctx).Info("Reconciling")
+
+		fn := func(c context.Context) (ctrl.Result, error) {
+			p := &corev1.Pod{}
+			if err := r.Get(ctx, req.NamespacedName, p); err != nil {
+				return ctrl.Result{}, staserrors.Ignore(err, apierrors.IsNotFound)
+			}
+
+			var stateWaiting *corev1.ContainerStateWaiting
+
+			for _, cs := range p.Status.ContainerStatuses {
+				if csw := cs.State.Waiting; csw != nil {
+					if _, ok := backoffContainerStateReasons[csw.Reason]; ok {
+						stateWaiting = csw
+						break
+					}
+				}
+			}
+
+			if stateWaiting == nil {
+				reasons := make([]string, 0, len(backoffContainerStateReasons))
+				for k := range backoffContainerStateReasons {
+					reasons = append(reasons, k)
+				}
+				return ctrl.Result{}, fmt.Errorf("no container-state waiting found with reasons %q in pod %q", reasons, p.Name)
+			}
+
+			pc := metav1.GetControllerOf(p)
+			if pc.Kind != "Job" {
```

Comment:
Should we add an additional condition to verify the namespace as we are only interested in jobs running in `image-operator-namespace`?

Reply:
I see that the namespace predicate takes care of it

Reply:
The namespace is the most important predicate. We could add a test for it, but I think it is not required....